plugin/src/main/java/org/opensearch/ml/action/agents/DeleteAgentTransportAction.java

```diff
@@ -30,41 +41,79 @@ public class DeleteAgentTransportAction extends HandledTransportAction<ActionReq
     Client client;
     NamedXContentRegistry xContentRegistry;
 
+    ClusterService clusterService;
+
     @Inject
     public DeleteAgentTransportAction(
         TransportService transportService,
         ActionFilters actionFilters,
         Client client,
-        NamedXContentRegistry xContentRegistry
+        NamedXContentRegistry xContentRegistry,
+        ClusterService clusterService
     ) {
         super(MLAgentDeleteAction.NAME, transportService, actionFilters, MLAgentDeleteRequest::new);
         this.client = client;
         this.xContentRegistry = xContentRegistry;
+        this.clusterService = clusterService;
     }
 
     @Override
     protected void doExecute(Task task, ActionRequest request, ActionListener<DeleteResponse> actionListener) {
         MLAgentDeleteRequest mlAgentDeleteRequest = MLAgentDeleteRequest.fromActionRequest(request);
         String agentId = mlAgentDeleteRequest.getAgentId();
-        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
-            ActionListener<DeleteResponse> wrappedListener = ActionListener.runBefore(actionListener, () -> context.restore());
-            DeleteRequest deleteRequest = new DeleteRequest(ML_AGENT_INDEX, agentId);
-            client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
-                @Override
-                public void onResponse(DeleteResponse deleteResponse) {
-                    log.debug("Completed Delete Agent Request, agent id:{} deleted", agentId);
-                    wrappedListener.onResponse(deleteResponse);
-                }
+        GetRequest getRequest = new GetRequest(ML_AGENT_INDEX).id(agentId);
+        boolean isSuperAdmin = isSuperAdminUserWrapper(clusterService, client);
 
-                @Override
-                public void onFailure(Exception e) {
-                    log.error("Failed to delete ML Agent " + agentId, e);
-                    wrappedListener.onFailure(e);
+        try (ThreadContext.StoredContext context = client.threadPool().getThreadContext().stashContext()) {
+            client.get(getRequest, ActionListener.wrap(getResponse -> {
+                if (getResponse.isExists()) {
```

Comment:
Nit: Nested if conditions makes it harder to read. I prefer following coding pattern to prevent nesting:

```
if (!getResponse.isExists()) {
  actionListener.onFailure(..);
  return;
}
try(...) {
// remaining code
}
```